WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -1,26 +1,100 @@
 using System;
+using System.Runtime.InteropServices;
 using System.Threading;
 using System.Threading.Tasks;
 using WalletWasabi.Bases;
 using WalletWasabi.Helpers;
+using WalletWasabi.Helpers.PowerSaving;
+using WalletWasabi.Logging;
 using WalletWasabi.Wallets;
+using static WalletWasabi.Helpers.PowerSaving.LinuxInhibitorTask;
 
 namespace WalletWasabi.Services
 {
 	public class SystemAwakeChecker : PeriodicRunner
 	{
-		public SystemAwakeChecker(WalletManager walletManager) : base(TimeSpan.FromMinutes(1))
+		private const string Reason = "CoinJoin is in progress.";
+		private static readonly TimeSpan Timeout = TimeSpan.FromMinutes(5);
+
+		private volatile IPowerSavingInhibitorTask? _powerSavingTask;
+
+		private SystemAwakeChecker(WalletManager walletManager, Func<Task<IPowerSavingInhibitorTask>>? taskFactory) : base(TimeSpan.FromMinutes(1))
 		{
 			WalletManager = walletManager;
+			TaskFactory = taskFactory;
 		}
 
 		private WalletManager WalletManager { get; }
+		public Func<Task<IPowerSavingInhibitorTask>>? TaskFactory { get; }
+
+		/// <summary>Checks whether we support awake state prolonging for the current platform.</summary>
+		public static async Task<SystemAwakeChecker?> CreateAsync(WalletManager walletManager)
+		{
+			Func<Task<IPowerSavingInhibitorTask>>? taskFactory = null;
+
+			if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+			{
+				// Either we have systemd or not.
+				bool isSystemd = await LinuxInhibitorTask.IsSystemdInhibitSupportedAsync().ConfigureAwait(false);
```

Comment:
"Name can be simplified"

Reply:
I don't want to simplify it too much because then it's just `Create` and you can create wrong task and you must check it in IDE whether it's really correct. That's the reason really why I didn't simplify it in the first place.

Reply:
ACK